In [1]:
import pandas as pd

In [2]:
EPWORTH = pd.read_csv("Epworth_Sleepiness_Scale.csv")

In [3]:
EPWORTH[EPWORTH['INFODT'] != EPWORTH['ORIG_ENTRY']]

REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  PTCGBOTH  ESS1  \
20    254953601        S   3100       BL  EPWORTH  10/2010       1.0     0   
21    262140401        S   3102       BL  EPWORTH  11/2010       1.0     3   
35    254639601        S   3551       BL  EPWORTH  10/2010       1.0     0   
55    278022401        S   3651       BL  EPWORTH  02/2011       1.0     1   
59    280555301        S   3557       BL  EPWORTH  12/2010       3.0     3   
...         ...      ...    ...      ...      ...      ...       ...   ...   
8222  770475501        S  52146      V10  EPWORTH  03/2020       1.0     2   
8223  770912901        S   3206      V16  EPWORTH  03/2020       1.0     1   
8224  771898601        S  60045      V13  EPWORTH  03/2020       3.0     0   
8225  772105801        S  56789      V08  EPWORTH  03/2020       1.0     0   
8226  772122001        S  52274      V10  EPWORTH  03/2020       1.0     1   

      ESS2  ESS3  ESS4  ESS5  ESS6  ESS7  ESS8 ORIG_ENTRY  \
20     1.0     1     1   3.0     0     1   0.0    11/2010   
21     2.0     1     2   2.0     1     2   1.0    12/2010   
35     1.0     0     0   1.0     0     0   0.0    11/2010   
55     1.0     2     1   3.0     0     0   0.0    03/2011   
59     3.0     2     2   3.0     0     2   0.0    03/2011   
...    ...   ...   ...   ...   ...   ...   ...        ...   
8222   2.0     0     1   2.0     0     0   0.0    04/2020   
8223   1.0     1     1   1.0     0     0   0.0    04/2020   
8224   3.0     0     2   3.0     0     3   0.0    06/2020   
8225   1.0     0     1   1.0     0     0   0.0    06/2020   
8226   0.0     0     0   0.0     0     0   0.0    06/2020   

                LAST_UPDATE  QUERY SITE_APRV  
20    2020-05-29 09:11:10.0    NaN   12/2010  
21    2020-01-02 13:27:10.0    NaN   12/2010  
35    2020-06-30 12:10:05.0    NaN   01/2011  
55    2020-01-17 08:30:46.0    NaN   03/2011  
59    2020-06-30 12:11:26.0    NaN   03/2011  
...                     ...    ...       ...  
8222  2020-06-24 19:16:44.0    NaN   04/2020  
8223  2020-06-26 19:15:08.0    NaN   04/2020  
8224  2020-06-30 16:55:43.0    NaN   06/2020  
8225  2020-06-30 16:13:07.0    NaN   06/2020  
8226  2020-06-30 15:47:13.0    NaN   06/2020  

[2420 rows x 19 columns]

In [4]:
Status = pd.read_csv("Patient_Status.csv")

In [5]:
Status

PATNO RECRUITMENT_CAT IMAGING_CAT ENROLL_DATE ENROLL_CAT ENROLL_STATUS  \
0     10362        PRODROMA    PRODROMA         NaN        NaN      Excluded   
1     10405        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2     10496        PRODROMA    PRODROMA         NaN        NaN      Excluded   
3     10606        PRODROMA    no image         NaN        NaN      Declined   
4     10662        PRODROMA    PRODROMA         NaN        NaN      Excluded   
...     ...             ...         ...         ...        ...           ...   
2246  92490        PRODROMA    PRODROMA     01/2015   PRODROMA      Enrolled   
2247  92516        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2248  92560        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2249  92834        PRODROMA    PRODROMA     02/2015   PRODROMA      Enrolled   
2250  92876        PRODROMA    PRODROMA         NaN        NaN      Excluded   

     DESCRP_CAT STATUS_DATE  
0           HYP     08/2013  
1           HYP     09/2014  
2           HYP     06/2013  
3           HYP     08/2013  
4           HYP     06/2014  
...         ...         ...  
2246        HYP     01/2015  
2247        HYP     11/2014  
2248        HYP     06/2014  
2249        HYP     02/2015  
2250        HYP     11/2014  

[2251 rows x 8 columns]

In [6]:
EPWORTH[EPWORTH['PATNO'] == 92490]

REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  PTCGBOTH  ESS1  \
2773  491661401        S  92490       BL  EPWORTH  01/2015       1.0     1   
3332  527236701        S  92490      V02  EPWORTH  07/2015       1.0     1   
3727  549525401        S  92490      V04  EPWORTH  01/2016       1.0     1   
4689  611445301        S  92490      V06  EPWORTH  01/2017       1.0     1   
6913  726992701        S  92490      V10  EPWORTH  01/2019       1.0     1   
8071  766305101        S  92490      V12  EPWORTH  01/2020       1.0     0   

      ESS2  ESS3  ESS4  ESS5  ESS6  ESS7  ESS8 ORIG_ENTRY  \
2773   1.0     1     0   3.0     0     1   0.0    01/2015   
3332   1.0     1     1   1.0     0     1   0.0    07/2015   
3727   1.0     1     0   1.0     0     1   0.0    01/2016   
4689   1.0     1     1   2.0     0     1   0.0    01/2017   
6913   0.0     0     1   1.0     0     0   0.0    01/2019   
8071   0.0     0     0   1.0     0     0   0.0    02/2020   

                LAST_UPDATE  QUERY SITE_APRV  
2773  2020-06-30 12:44:17.0    NaN   01/2015  
3332  2020-06-30 12:44:19.0    NaN   08/2015  
3727  2020-06-30 12:44:19.0    NaN   01/2016  
4689  2020-06-30 12:44:20.0    NaN   01/2017  
6913  2020-06-30 12:44:21.0    NaN   01/2019  
8071  2020-06-30 12:44:21.0    NaN   02/2020

In [7]:
Status[Status['PATNO'] == 92490]

PATNO RECRUITMENT_CAT IMAGING_CAT ENROLL_DATE ENROLL_CAT ENROLL_STATUS  \
2246  92490        PRODROMA    PRODROMA     01/2015   PRODROMA      Enrolled   

     DESCRP_CAT STATUS_DATE  
2246        HYP     01/2015

In [8]:
MDS1 = pd.read_csv("MDS_UPDRS_Part_I.csv")

In [9]:
MDS1[MDS1['PATNO'] == 92490]

REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  NUPSOURC  NP1COG  \
5188   489538101        S  92490       SC  NUPDRS1  12/2014         1       0   
5247   491653201        S  92490       BL  NUPDRS1  01/2015         1       1   
5780   520804901        S  92490      V01  NUPDRS1  05/2015         1       1   
6190   527235901        S  92490      V02  NUPDRS1  07/2015         1       0   
6455   538794401        S  92490      V03  NUPDRS1  10/2015         1       1   
6946   549517301        S  92490      V04  NUPDRS1  01/2016         1       1   
7796   579166001        S  92490      V05  NUPDRS1  07/2016         1       0   
8668   611439501        S  92490      V06  NUPDRS1  01/2017         1       0   
9560   643924101        S  92490      V07  NUPDRS1  07/2017         1       0   
12017  726984201        S  92490      V10  NUPDRS1  01/2019         1       0   
12864  750470901        S  92490      V11  NUPDRS1  07/2019         1       1   
13670  766077101        S  92490      V12  NUPDRS1  01/2020         1       1   

       NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS ORIG_ENTRY  \
5188         0        0      0.0      0.0     0.0    12/2014   
5247         0        0      0.0      2.0     0.0    01/2015   
5780         0        0      1.0      0.0     0.0    05/2015   
6190         0        0      0.0      1.0     0.0    07/2015   
6455         0        0      0.0      0.0     0.0    10/2015   
6946         0        0      0.0      2.0     0.0    01/2016   
7796         0        0      0.0      1.0     0.0    07/2016   
8668         0        0      0.0      0.0     0.0    01/2017   
9560         0        0      0.0      0.0     0.0    07/2017   
12017        0        0      0.0      1.0     0.0    01/2019   
12864        0        0      0.0      0.0     0.0    07/2019   
13670        0        0      0.0      1.0     0.0    01/2020   

                 LAST_UPDATE  QUERY SITE_APRV  
5188   2020-06-30 12:44:18.0    NaN   12/2014  
5247   2020-06-30 12:44:17.0    NaN   01/2015  
5780   2020-06-30 12:44:19.0    NaN   05/2015  
6190   2020-06-30 12:44:19.0    NaN   08/2015  
6455   2020-06-30 12:44:19.0    NaN   10/2015  
6946   2020-06-30 12:44:19.0    NaN   01/2016  
7796   2020-06-30 12:44:20.0    NaN   07/2016  
8668   2020-06-30 12:44:20.0    NaN   01/2017  
9560   2020-06-30 12:44:20.0    NaN   07/2017  
12017  2020-06-30 12:44:21.0    NaN   01/2019  
12864  2020-06-30 12:44:21.0    NaN   07/2019  
13670  2020-06-30 12:44:21.0    NaN   01/2020

In [10]:
MDS1[MDS1['EVENT_ID'] == 'BL'][MDS1['INFODT'] != MDS1['ORIG_ENTRY']]

C:\Users\brian\anaconda3\lib\site-packages\ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  NUPSOURC  NP1COG  \
41     254948901        S   3100       BL  NUPDRS1  10/2010         1       0   
48     262133601        S   3102       BL  NUPDRS1  11/2010         3       0   
64     254632501        S   3551       BL  NUPDRS1  10/2010         1       0   
106    278011301        S   3651       BL  NUPDRS1  02/2011         1       0   
131    280285001        S   3557       BL  NUPDRS1  12/2010         1       1   
...          ...      ...    ...      ...      ...      ...       ...     ...   
12911  751588801        S  41513       BL  NUPDRS1  12/2018         1       0   
13005  753965401        S  40750       BL  NUPDRS1  07/2019         3       1   
13023  754293601        S  40767       BL  NUPDRS1  06/2019         3       1   
13035  754480101        S  43086       BL  NUPDRS1  06/2019         1       0   
13036  754488301        S  42381       BL  NUPDRS1  05/2019         1       0   

       NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS ORIG_ENTRY  \
41           0        0      0.0      0.0     0.0    11/2010   
48           0        1      0.0      0.0     0.0    12/2010   
64           0        0      0.0      0.0     0.0    11/2010   
106          0        0      0.0      0.0     0.0    03/2011   
131          1        0      1.0      2.0     0.0    03/2011   
...        ...      ...      ...      ...     ...        ...   
12911        0        0      0.0      0.0     0.0    08/2019   
13005        0        1      1.0      0.0     0.0    09/2019   
13023        0        2      2.0      0.0     0.0    09/2019   
13035        0        0      2.0      0.0     0.0    09/2019   
13036        0        0      1.0      0.0     0.0    09/2019   

                 LAST_UPDATE  QUERY SITE_APRV  
41     2020-05-29 09:11:10.0    NaN   12/2010  
48     2020-01-02 13:27:10.0    NaN   12/2010  
64     2020-06-30 12:10:05.0    NaN   01/2011  
106    2020-01-17 08:30:47.0    NaN   03/2011  
131    2020-06-30 12:11:26.0    NaN   03/2011  
...                      ...    ...       ...  
12911  2020-06-30 17:23:15.0    NaN   08/2019  
13005  2020-06-30 17:20:12.0    NaN   09/2019  
13023  2020-06-30 17:20:13.0    NaN   09/2019  
13035  2020-06-30 17:20:15.0    NaN   09/2019  
13036  2020-06-23 16:35:50.0    NaN   09/2019  

[641 rows x 17 columns]

In [11]:
Status[Status['PATNO'] == 3102]

PATNO RECRUITMENT_CAT IMAGING_CAT ENROLL_DATE ENROLL_CAT ENROLL_STATUS  \
157   3102              PD          PD     11/2010         PD      Withdrew   

    DESCRP_CAT STATUS_DATE  
157        NaN     04/2016

In [12]:
MDS1[MDS1['PATNO'] == 3102]

REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  NUPSOURC  NP1COG  \
43    255294201        S   3102       SC  NUPDRS1  10/2010         3       0   
48    262133601        S   3102       BL  NUPDRS1  11/2010         3       0   
102   277807601        S   3102      V01  NUPDRS1  02/2011         3       0   
287   297102601        S   3102      V02  NUPDRS1  05/2011         1       0   
479   312326201        S   3102      V03  NUPDRS1  09/2011         1       0   
761   321952801        S   3102       ST  NUPDRS1  11/2011         3       1   
1193  347943301        S   3102      V05  NUPDRS1  05/2012         1       2   
2347  385324901        S   3102      V06  NUPDRS1  02/2013         3       2   
2754  396646501        S   3102      V07  NUPDRS1  04/2013         1       2   
3620  420441701        S   3102      V08  NUPDRS1  11/2013         1       2   
4173  437848701        S   3102      V09  NUPDRS1  05/2014         1       2   
4987  476354801        S   3102      V10  NUPDRS1  11/2014         1       2   
5775  520756101        S   3102      V11  NUPDRS1  04/2015         1       2   

      NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS ORIG_ENTRY  \
43          0        0      0.0      0.0     0.0    11/2010   
48          0        1      0.0      0.0     0.0    12/2010   
102         0        0      0.0      0.0     0.0    02/2011   
287         0        1      0.0      1.0     0.0    06/2011   
479         0        1      0.0      1.0     0.0    09/2011   
761         0        0      0.0      0.0     0.0    11/2011   
1193        0        1      0.0      2.0     0.0    05/2012   
2347        2        1      0.0      2.0     0.0    02/2013   
2754        1        1      0.0      2.0     0.0    05/2013   
3620        1        2      1.0      2.0     0.0    12/2013   
4173        1        2      1.0      2.0     0.0    05/2014   
4987        1        1      1.0      1.0     0.0    11/2014   
5775        1        1      1.0      1.0     0.0    05/2015   

                LAST_UPDATE  QUERY SITE_APRV  
43    2020-01-02 13:27:11.0    NaN   12/2010  
48    2020-01-02 13:27:10.0    NaN   12/2010  
102   2020-01-02 13:27:12.0    NaN   03/2011  
287   2020-01-02 13:27:13.0    NaN   06/2011  
479   2020-01-02 13:27:13.0    NaN   09/2011  
761   2020-01-02 13:27:12.0    NaN   12/2011  
1193  2020-01-02 13:27:13.0    NaN   05/2012  
2347  2020-01-02 13:27:13.0    NaN   02/2013  
2754  2020-01-02 13:27:13.0    NaN   05/2013  
3620  2020-01-02 13:27:14.0    NaN   12/2013  
4173  2020-01-02 13:27:14.0    NaN   05/2014  
4987  2020-01-02 13:27:14.0    NaN   11/2014  
5775  2020-01-02 13:27:15.0    NaN   05/2015

각 데이터 파일에서 EVENT_ID가 'BL'인 경우에만 Patient_Status가 존재, 날짜는 각 데이터 파일의 'INFODT'와 Status파일의 'ENROLL_DATE'로 일치.

# 데이터셋 통합

In [13]:
EPWORTH = pd.read_csv("Epworth_Sleepiness_Scale.csv")
ARM = pd.read_csv("Gait_Data___Arm_swing.csv")
LEXTREMITY = pd.read_csv("Lower_Extremity_Function__Mobility_.csv")
MDS1 = pd.read_csv("MDS_UPDRS_Part_I.csv")
MDS1Q = pd.read_csv("MDS_UPDRS_Part_I__Patient_Questionnaire.csv")
MDS2Q = pd.read_csv("MDS_UPDRS_Part_II__Patient_Questionnaire.csv")
SCHWAB = pd.read_csv("Modified_Schwab_+_England_ADL.csv")
HOUSEHOLD = pd.read_csv("PASE_-_Household_Activity.csv")
LEISURE = pd.read_csv("PASE_-_Leisure_Time_Activity.csv")
STATUS = pd.read_csv("Patient_Status.csv")
REM = pd.read_csv("REM_Sleep_Disorder_Questionnaire.csv")
TAPPD = pd.read_csv("TAP-PD_Kinetics_Device_Testing.csv")
UEXTREMITY = pd.read_csv("Upper_Extremity_Function-_Short_Form.csv")

In [14]:
ARM[ARM['EVENT_ID']=="BL"]
# 해당 데이터 크기가 너무 작음...

PATNO EVENT_ID   INFODT  COHORT   SP_U   RA_AMP_U   LA_AMP_U  RA_STD_U  \
4    42426       BL  11/2016     1.0  0.982  51.516231  30.989870  7.412588   
6    42422       BL  11/2016     1.0  1.143  38.314673  33.248165  7.560887   
9    42418       BL  02/2017     3.0  0.964  15.610665  36.560067  4.258304   
12   42357       BL  01/2017     1.0  1.050  33.746782   5.899303  5.797351   
13   42415       BL  12/2016     3.0  0.995  20.253311  25.345497  2.601230   
15   42351       BL  11/2016     3.0  0.741   6.593724  10.045577  1.749566   
18   42346       BL  11/2016     3.0  1.169  41.221711  11.183933  1.789196   
32   41342       BL  11/2016     1.0  1.067  19.362595  29.414106  4.509690   
36   41299       BL  12/2016     1.0  1.418        NaN        NaN       NaN   
85   42035       BL  03/2019     3.0  0.962  27.731571   7.483832  4.440230   
86   42034       BL  02/2019     3.0  0.870  10.656894  42.200638  2.758563   
88   42014       BL  04/2017     3.0  0.972  22.044892  26.202028  2.929272   
90   42012       BL  03/2017     1.0  1.164  16.849540  31.536668  1.433675   
130  40567       BL  02/2019     1.0  0.973  29.782323  41.685751  9.155024   
145  42444       BL  11/2016     1.0  1.372  50.588478  50.846473  3.409763   
148  42445       BL  11/2016     1.0  1.618  51.195557  44.207813  8.932798   
149  42446       BL  11/2016     1.0  1.050  47.633597  35.224929  4.842927   
152  42447       BL  11/2016     3.0  1.272  20.811657  20.972986  2.774855   
155  42448       BL  12/2016     1.0  1.242  50.928350  22.036564  6.083692   
157  42450       BL  12/2016     1.0  1.433  36.878608  33.377743  4.481835   
160  42451       BL  12/2016     1.0  1.333  30.460233  70.541678  6.811038   
161  42452       BL  12/2016     1.0  1.271  16.311517  40.832855  6.109233   
162  42453       BL  12/2016     3.0  0.982   2.764410  25.210230  1.611778   
163  42455       BL  01/2017     3.0  1.229        NaN        NaN       NaN   
164  42456       BL  02/2017     3.0  1.307  40.041035  32.668807  3.980181   
186  42033       BL  05/2019     NaN    NaN  17.210730  21.075388  3.104887   

      LA_STD_U     SYM_U  ...  TUG1_STRAIGHT_DUR  TUG1_TURNS_DUR  \
4     4.762775  0.672047  ...           0.632812        1.710938   
6     8.128498  0.168624  ...           0.625000        1.437500   
9     3.149224  0.569155  ...           0.617188        2.472656   
12    2.274804  5.164517  ...           0.515625        1.453125   
13    3.655499  0.198092  ...           0.570312        2.699219   
15    2.098780  0.339688  ...           0.656250        1.894531   
18    3.223660  2.800767  ...           0.570312        1.347656   
32    2.984660  0.342442  ...           0.515625        1.207031   
36         NaN       NaN  ...           0.390625        1.984375   
85    2.576723  2.953617  ...           0.585938        1.597656   
86    4.398527  0.744203  ...           0.554688        2.214844   
88    3.886554  0.171402  ...           0.648438        2.332031   
90    2.169746  0.466460  ...           0.601562        2.171875   
130  18.263038  0.811041  ...           0.601562        1.753906   
145   6.415280  0.199265  ...           0.515625        1.187500   
148   8.807786  0.794208  ...           0.546875        1.234375   
149   2.469952  0.352738  ...           0.546875        1.980469   
152   3.700150  0.496404  ...           0.531250        1.738281   
155   4.609800  1.326843  ...           0.500000        1.160156   
157   3.457867  0.109992  ...           0.578125        1.144531   
160  13.168107  0.558768  ...           0.539062        1.691406   
161   4.708778  0.596745  ...           0.710938        1.882812   
162   3.438281  0.890630  ...           0.609375        2.347656   
163        NaN       NaN  ...           0.640625        2.160156   
164   4.627993  0.237751  ...           0.632812        1.953125   
186   1.238529  0.327869  ...           0.804688        1.964844   

     TUG1_STEP_REG  TUG1_STEP_SYM 

In [15]:
LEXTREMITY # Baseline 자체가 존재하지 않음

REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  NQMOB37  NQMOB30  \
0    723072701        S   4101      V13  NQOLLEFS  01/2019        4        4   
1    723251801        S   3522      V14  NQOLLEFS  01/2019        5        5   
2    723412901        S   3203      V15  NQOLLEFS  01/2019        5        4   
3    723823401        S   3479      V13  NQOLLEFS  01/2019        5        5   
4    765526501        S   3173      V15  NQOLLEFS  01/2020        5        5   
..         ...      ...    ...      ...       ...      ...      ...      ...   
427  770246201        S   4033      V14  NQOLLEFS  02/2020        5        5   
428  770293601        S   4035      V14  NQOLLEFS  03/2020        5        5   
429  770345401        S   3013      V15  NQOLLEFS  12/2019        5        5   
430  770915501        S   3206      V16  NQOLLEFS  03/2020        5        5   
431  771895001        S  60045      V13  NQOLLEFS  03/2020        5        3   

     NQMOB26  NQMOB32  NQMOB25  NQMOB33  NQMOB31  NQMOB28 ORIG_ENTRY  \
0          4        4        2        4        4        4    01/2019   
1          5        5        5        5        5        5    01/2019   
2          5        5        5        5        5        5    01/2019   
3          5        5        5        5        5        5    01/2019   
4          4        5        4        5        3        5    01/2020   
..       ...      ...      ...      ...      ...      ...        ...   
427        5        5        5        5        5        5    04/2020   
428        5        5        5        5        5        5    04/2020   
429        5        5        5        5        5        5    04/2020   
430        5        5        5        5        5        5    04/2020   
431        4        4        5        5        3        5    06/2020   

               LAST_UPDATE  QUERY SITE_APRV  
0    2020-06-29 14:26:27.0    NaN   01/2019  
1    2020-06-25 15:26:13.0    NaN   01/2019  
2    2020-06-26 19:14:44.0    NaN   01/2019  
3    2020-06-24 19:05:57.0    NaN   01/2019  
4    2020-06-30 09:46:22.0    NaN   01/2020  
..                     ...    ...       ...  
427  2020-06-29 13:59:17.0    NaN   04/2020  
428  2020-06-29 14:02:47.0    NaN   04/2020  
429  2020-06-30 09:26:13.0    NaN   04/2020  
430  2020-06-26 19:15:09.0    NaN   04/2020  
431  2020-06-30 16:55:43.0    NaN   06/2020  

[432 rows x 18 columns]

In [16]:
MDS1[MDS1['EVENT_ID']=="BL"]

REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME   INFODT  NUPSOURC  NP1COG  \
1      233164301        S   3400       BL  NUPDRS1  07/2010         1       0   
3      234837501        S   3404       BL  NUPDRS1  07/2010         1       0   
6      232528701        S   3401       BL  NUPDRS1  07/2010         1       0   
7      231737401        S   3403       BL  NUPDRS1  07/2010         1       0   
8      236118001        S   3405       BL  NUPDRS1  07/2010         1       0   
...          ...      ...    ...      ...      ...      ...       ...     ...   
13329  759462901        S   5012       BL  NUPDRS1  11/2019         1       1   
13592  764394701        S   5014       BL  NUPDRS1  01/2020         1       0   
13594  764423801        S   5015       BL  NUPDRS1  01/2020         1       2   
13613  764757401        S   5016       BL  NUPDRS1  01/2020         1       2   
13701  766591701        S   5017       BL  NUPDRS1  02/2020         1       0   

       NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS ORIG_ENTRY  \
1            0        2      3.0      2.0     0.0    07/2010   
3            0        0      0.0      0.0     0.0    07/2010   
6            0        0      0.0      0.0     0.0    07/2010   
7            0        0      0.0      0.0     0.0    07/2010   
8            0        0      0.0      0.0     0.0    07/2010   
...        ...      ...      ...      ...     ...        ...   
13329        0        0      1.0      2.0     0.0    11/2019   
13592        0        0      1.0      0.0     0.0    01/2020   
13594        0        0      0.0      1.0     0.0    01/2020   
13613        0        1      1.0      0.0     0.0    01/2020   
13701        0        0      0.0      0.0     0.0    02/2020   

                 LAST_UPDATE  QUERY SITE_APRV  
1      2020-06-29 14:40:01.0    NaN   07/2010  
3      2020-06-29 14:41:41.0    NaN   07/2010  
6      2020-06-29 14:40:25.0    NaN   07/2010  
7      2020-06-29 14:40:44.0    NaN   07/2010  
8      2020-01-13 15:16:51.0    NaN   07/2010  
...                      ...    ...       ...  
13329  2020-05-01 14:36:50.0    NaN   11/2019  
13592  2020-05-01 14:40:10.0    NaN   01/2020  
13594  2020-05-01 14:41:02.0    NaN   01/2020  
13613  2020-05-01 14:41:45.0    NaN   01/2020  
13701  2020-05-01 14:43:27.0    NaN   02/2020  

[1923 rows x 17 columns]

In [17]:
MDS1Q[MDS1Q['EVENT_ID']=="BL"]

REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  NUPSOURC  \
2      233164501        S   3400       BL  NUPDRS1P  07/2010         1   
3      234837801        S   3404       BL  NUPDRS1P  07/2010         1   
4      232529501        S   3401       BL  NUPDRS1P  07/2010         1   
7      236118201        S   3405       BL  NUPDRS1P  07/2010         1   
8      231716301        S   3403       BL  NUPDRS1P  07/2010         1   
...          ...      ...    ...      ...       ...      ...       ...   
13331  759463301        S   5012       BL  NUPDRS1P  11/2019         1   
13595  764394901        S   5014       BL  NUPDRS1P  01/2020         1   
13598  764424801        S   5015       BL  NUPDRS1P  01/2020         1   
13616  764758301        S   5016       BL  NUPDRS1P  01/2020         1   
13703  766593501        S   5017       BL  NUPDRS1P  02/2020         1   

       NP1SLPN  NP1SLPD  NP1PAIN  NP1URIN  NP1CNST  NP1LTHD  NP1FATG  \
2          0.0      1.0      2.0      1.0      1.0      1.0      4.0   
3          1.0      1.0      0.0      1.0      0.0      0.0      1.0   
4          0.0      0.0      0.0      0.0      0.0      0.0      0.0   
7          0.0      0.0      0.0      0.0      0.0      0.0      0.0   
8          2.0      2.0      3.0      1.0      2.0      0.0      1.0   
...        ...      ...      ...      ...      ...      ...      ...   
13331      0.0      1.0      0.0      3.0      3.0      1.0      1.0   
13595      2.0      0.0      2.0      3.0      2.0      1.0      1.0   
13598      1.0      2.0      3.0      0.0      0.0      0.0      1.0   
13616      0.0      0.0      1.0      0.0      0.0      0.0      0.0   
13703      1.0      0.0      0.0      0.0      0.0      0.0      0.0   

      ORIG_ENTRY            LAST_UPDATE  QUERY SITE_APRV  
2        07/2010  2020-06-29 14:40:01.0    NaN   07/2010  
3        07/2010  2020-06-29 14:41:41.0    NaN   07/2010  
4        07/2010  2020-06-29 14:40:25.0    NaN   07/2010  
7        07/2010  2020-01-13 15:16:51.0    NaN   07/2010  
8        07/2010  2020-06-29 14:40:44.0    NaN   07/2010  
...          ...                    ...    ...       ...  
13331    11/2019  2020-05-01 14:36:49.0    NaN   11/2019  
13595    01/2020  2020-05-01 14:40:10.0    NaN   01/2020  
13598    01/2020  2020-05-01 14:41:02.0    NaN   01/2020  
13616    01/2020  2020-05-01 14:41:45.0    NaN   01/2020  
13703    02/2020  2020-05-01 14:43:28.0    NaN   02/2020  

[1924 rows x 18 columns]

In [18]:
MDS1 = MDS1[MDS1['EVENT_ID']=="BL"]
MDS1Q = MDS1Q[MDS1Q['EVENT_ID']=="BL"]

In [19]:
del MDS1['EVENT_ID'], MDS1Q['EVENT_ID']
del MDS1['REC_ID'], MDS1Q['REC_ID']
del MDS1['F_STATUS'], MDS1Q['F_STATUS']
del MDS1['QUERY'], MDS1Q['QUERY']
del MDS1['ORIG_ENTRY'], MDS1Q['ORIG_ENTRY']
del MDS1['PAG_NAME'], MDS1Q['PAG_NAME']
del MDS1['INFODT'], MDS1Q['INFODT']
del MDS1['SITE_APRV'], MDS1Q['SITE_APRV']
del MDS1['LAST_UPDATE'], MDS1Q['LAST_UPDATE']

In [20]:
MDS1

PATNO  NUPSOURC  NP1COG  NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS
1       3400         1       0        0        2      3.0      2.0     0.0
3       3404         1       0        0        0      0.0      0.0     0.0
6       3401         1       0        0        0      0.0      0.0     0.0
7       3403         1       0        0        0      0.0      0.0     0.0
8       3405         1       0        0        0      0.0      0.0     0.0
...      ...       ...     ...      ...      ...      ...      ...     ...
13329   5012         1       1        0        0      1.0      2.0     0.0
13592   5014         1       0        0        0      1.0      0.0     0.0
13594   5015         1       2        0        0      0.0      1.0     0.0
13613   5016         1       2        0        1      1.0      0.0     0.0
13701   5017         1       0        0        0      0.0      0.0     0.0

[1923 rows x 8 columns]

In [21]:
MDS1Q

PATNO  NUPSOURC  NP1SLPN  NP1SLPD  NP1PAIN  NP1URIN  NP1CNST  NP1LTHD  \
2       3400         1      0.0      1.0      2.0      1.0      1.0      1.0   
3       3404         1      1.0      1.0      0.0      1.0      0.0      0.0   
4       3401         1      0.0      0.0      0.0      0.0      0.0      0.0   
7       3405         1      0.0      0.0      0.0      0.0      0.0      0.0   
8       3403         1      2.0      2.0      3.0      1.0      2.0      0.0   
...      ...       ...      ...      ...      ...      ...      ...      ...   
13331   5012         1      0.0      1.0      0.0      3.0      3.0      1.0   
13595   5014         1      2.0      0.0      2.0      3.0      2.0      1.0   
13598   5015         1      1.0      2.0      3.0      0.0      0.0      0.0   
13616   5016         1      0.0      0.0      1.0      0.0      0.0      0.0   
13703   5017         1      1.0      0.0      0.0      0.0      0.0      0.0   

       NP1FATG  
2          4.0  
3          1.0  
4          0.0  
7          0.0  
8          1.0  
...        ...  
13331      1.0  
13595      1.0  
13598      1.0  
13616      0.0  
13703      0.0  

[1924 rows x 9 columns]

In [22]:
MDS2Q = MDS2Q[MDS2Q['EVENT_ID']=="BL"]

In [23]:
del MDS2Q['REC_ID'], MDS2Q['F_STATUS'],MDS2Q['EVENT_ID'],MDS2Q['PAG_NAME'], MDS2Q['INFODT']
del MDS2Q['ORIG_ENTRY'], MDS2Q['LAST_UPDATE'], MDS2Q['QUERY'], MDS2Q['SITE_APRV']
MDS2Q

PATNO  NUPSOURC  NP2SPCH  NP2SALV  NP2SWAL  NP2EAT  NP2DRES  NP2HYGN  \
0       3401         1        0      0.0      0.0     0.0      0.0      0.0   
5       3404         1        0      0.0      0.0     0.0      0.0      0.0   
6       3400         1        0      2.0      0.0     2.0      1.0      1.0   
7       3405         1        0      0.0      0.0     0.0      0.0      0.0   
8       3403         1        0      0.0      0.0     1.0      1.0      0.0   
...      ...       ...      ...      ...      ...     ...      ...      ...   
13333   5012         1        0      3.0      0.0     1.0      2.0      0.0   
13596   5014         1        1      0.0      0.0     0.0      1.0      0.0   
13599   5015         1        1      3.0      1.0     0.0      1.0      1.0   
13617   5016         1        0      0.0      0.0     0.0      0.0      0.0   
13705   5017         1        0      0.0      0.0     0.0      0.0      0.0   

       NP2HWRT  NP2HOBB  NP2TURN  NP2TRMR  NP2RISE  NP2WALK  NP2FREZ  
0          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
5          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
6          1.0      2.0      1.0      1.0      2.0      1.0      1.0  
7          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
8          1.0      1.0      1.0      0.0      2.0      1.0      0.0  
...        ...      ...      ...      ...      ...      ...      ...  
13333      3.0      4.0      2.0      2.0      3.0      1.0      2.0  
13596      0.0      0.0      1.0      0.0      1.0      1.0      0.0  
13599      1.0      1.0      1.0      1.0      2.0      3.0      0.0  
13617      0.0      4.0      0.0      2.0      4.0      3.0      0.0  
13705      0.0      0.0      0.0      1.0      0.0      0.0      0.0  

[1924 rows x 15 columns]

In [24]:
MDS2Q.dropna(how='any')

PATNO  NUPSOURC  NP2SPCH  NP2SALV  NP2SWAL  NP2EAT  NP2DRES  NP2HYGN  \
0       3401         1        0      0.0      0.0     0.0      0.0      0.0   
5       3404         1        0      0.0      0.0     0.0      0.0      0.0   
6       3400         1        0      2.0      0.0     2.0      1.0      1.0   
7       3405         1        0      0.0      0.0     0.0      0.0      0.0   
8       3403         1        0      0.0      0.0     1.0      1.0      0.0   
...      ...       ...      ...      ...      ...     ...      ...      ...   
13333   5012         1        0      3.0      0.0     1.0      2.0      0.0   
13596   5014         1        1      0.0      0.0     0.0      1.0      0.0   
13599   5015         1        1      3.0      1.0     0.0      1.0      1.0   
13617   5016         1        0      0.0      0.0     0.0      0.0      0.0   
13705   5017         1        0      0.0      0.0     0.0      0.0      0.0   

       NP2HWRT  NP2HOBB  NP2TURN  NP2TRMR  NP2RISE  NP2WALK  NP2FREZ  
0          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
5          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
6          1.0      2.0      1.0      1.0      2.0      1.0      1.0  
7          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
8          1.0      1.0      1.0      0.0      2.0      1.0      0.0  
...        ...      ...      ...      ...      ...      ...      ...  
13333      3.0      4.0      2.0      2.0      3.0      1.0      2.0  
13596      0.0      0.0      1.0      0.0      1.0      1.0      0.0  
13599      1.0      1.0      1.0      1.0      2.0      3.0      0.0  
13617      0.0      4.0      0.0      2.0      4.0      3.0      0.0  
13705      0.0      0.0      0.0      1.0      0.0      0.0      0.0  

[1924 rows x 15 columns]

In [25]:
SCHWAB = SCHWAB[SCHWAB['EVENT_ID']=='BL']

In [26]:
SCHWAB = SCHWAB.drop(['REC_ID', 'F_STATUS', 'EVENT_ID', 'PAG_NAME', 'INFODT',
       'ORIG_ENTRY', 'LAST_UPDATE', 'QUERY', 'SITE_APRV'], axis=1)
SCHWAB

PATNO  MSEADLG
3       3400       90
5       3403       80
6       3406       80
8       3407       90
11      3409       90
...      ...      ...
12172   5012       50
12414   5014      100
12416   5015       80
12437   5016       80
12519   5017      100

[1727 rows x 2 columns]

In [27]:
HOUSEHOLD = HOUSEHOLD[HOUSEHOLD['EVENT_ID']=='BL']

In [28]:
HOUSEHOLD = HOUSEHOLD.drop(['REC_ID', 'F_STATUS', 'EVENT_ID', 'PAG_NAME',
                      'ORIG_ENTRY', 'LAST_UPDATE', 'QUERY', 'SITE_APRV'], axis=1)

In [29]:
HOUSEHOLD = HOUSEHOLD.dropna(how='any')
HOUSEHOLD

PATNO  LTHSWRK  HVYHSWRK  HMREPR  LAWNWRK  OUTGARDN  CAREGVR  WRKVL  \
44     3134        2         2       1        2       1.0      2.0    2.0   
76    60015        2         2       2        2       2.0      1.0    2.0   
168   12224        2         2       1        2       2.0      1.0    2.0   
217   14281        2         2       1        1       1.0      1.0    2.0   
241   60023        2         2       1        1       1.0      2.0    2.0   
...     ...      ...       ...     ...      ...       ...      ...    ...   
4977  41513        1         1       1        1       1.0      1.0    2.0   
5025  40750        1         1       1        1       1.0      1.0    2.0   
5035  40767        2         2       1        2       2.0      1.0    2.0   
5042  43086        2         1       1        1       1.0      2.0    1.0   
5043  42381        2         1       1        1       2.0      1.0    2.0   

      WRKVLHR  WRKVLACT  
44       50.0       3.0  
76        2.0       3.0  
168       2.0       2.0  
217       2.0       1.0  
241      12.0       1.0  
...       ...       ...  
4977     30.0       1.0  
5025     20.0       1.0  
5035     50.0       4.0  
5042     40.0       1.0  
5043      3.0       1.0  

[457 rows x 10 columns]

In [30]:
LEISURE[LEISURE['EVENT_ID']=='BL'] # 곤란...

REC_ID F_STATUS  PATNO EVENT_ID PAG_NAME  QUESTNO  ACTVOFT  \
282    399545901        S   3134       BL     PASE        5      1.0   
283    399546001        S   3134       BL     PASE        6      1.0   
284    399545501        S   3134       BL     PASE        1      3.0   
285    399545601        S   3134       BL     PASE        2      3.0   
286    399545701        S   3134       BL     PASE        3      2.0   
...          ...      ...    ...      ...      ...      ...      ...   
33571  750819801        S  43080       BL     PASE        2      3.0   
33572  750819901        S  43080       BL     PASE        3      0.0   
33573  750820001        S  43080       BL     PASE        4      1.0   
33574  750820101        S  43080       BL     PASE        5      2.0   
33575  750820201        S  43080       BL     PASE        6      2.0   

              ACTVSPEC  HRDAYFRQ ORIG_ENTRY            LAST_UPDATE QUERY  \
282            jogging       1.0    05/2013  2020-06-01 15:32:13.0   NaN   
283        gym weights       1.0    05/2013  2020-06-01 15:32:13.0   NaN   
284                 tv       3.0    05/2013  2020-06-01 15:32:13.0   NaN   
285                NaN       2.0    05/2013  2020-06-01 15:32:13.0   NaN   
286         basketball       2.0    05/2013  2020-06-01 15:32:13.0   NaN   
...                ...       ...        ...                    ...   ...   
33571              NaN       1.0    07/2019  2020-06-23 16:36:42.0   NaN   
33572              NaN       NaN    07/2019  2020-06-23 16:36:42.0   NaN   
33573              box       2.0    07/2019  2020-06-23 16:36:42.0   NaN   
33574          jogging       1.0    07/2019  2020-06-23 16:36:42.0   NaN   
33575  lifting weights       1.0    07/2019  2020-06-23 16:36:42.0   NaN   

                   SITE_APRV  
282    2013-05-29 17:31:05.0  
283    2013-05-29 17:31:05.0  
284    2013-05-29 17:31:05.0  
285    2013-05-29 17:31:05.0  
286    2013-05-29 17:31:05.0  
...                      ...  
33571  2020-03-31 06:38:17.0  
33572  2020-03-31 06:38:17.0  
33573  2020-03-31 06:38:17.0  
33574  2020-03-31 06:38:17.0  
33575  2020-03-31 06:38:17.0  

[4488 rows x 13 columns]

In [31]:
TAPPD

PATNO TESTDATE  TAPDAY REVISION     DEVICESN  QMAT_OMS  QMAT_KEYDOWN  \
0     3432  09/2011     0.0       R4  HVDG9341016    40.462         3.717   
1     3432  09/2011     7.0       R4  HVDG9341016    41.180         3.590   
2     3432  09/2011    14.0       R4  HVDG9341016    41.975         3.506   
3     3435  10/2011     0.0       R4  HVDG9353016    46.176         3.765   
4     3218  10/2011     0.0       R4  HVDG9341013    39.231         4.139   
..     ...      ...     ...      ...          ...       ...           ...   
498   3088  04/2014   375.0       R4  HVDG9341020    31.757         4.285   
499   3089  04/2014   362.0       R4  HVDG9353019    32.667         3.908   
500   3234  06/2014   458.0       R4  HVDG9353015    34.305         4.063   
501   3086  10/2014     NaN       R4  HVDG9343006    22.716         4.747   
502   3086  03/2015     NaN       R4  HVDG9343006    25.106         4.772   

     QMAT_KEYDOWN_RIGHT  QMAT_KEYDOWN_LEFT  QMAT_PEGCYCLE  \
0                 3.633              3.793          0.808   
1                 3.495              3.677          0.750   
2                 3.356              3.636          0.704   
3                 3.438              4.012          1.002   
4                 4.464              3.655          0.941   
..                  ...                ...            ...   
498               4.492              4.025          0.676   
499               4.109              3.654          0.755   
500               4.189              3.920          0.734   
501               4.897              4.570          0.680   
502               4.897              4.630          0.826   

     QMAT_PEGCYCLE_RIGHT  QMAT_PEGCYCLE_LEFT  QMAT_KEYTRAN  \
0                  0.844               0.770        -0.980   
1                  0.785               0.713        -0.981   
2                  0.715               0.694        -0.904   
3                  1.072               0.927        -0.417   
4                  0.809               1.057        -0.578   
..                   ...                 ...           ...   
498                0.431               0.872        -0.766   
499                0.699               0.809        -1.772   
500                0.700               0.766        -1.025   
501                0.574               0.776        -1.296   
502                0.823               0.828        -1.246   

     QMAT_KEYTRAN_RIGHT  QMAT_KEYTRAN_LEFT DOM_SIDE AFF_SIDE     SEX  CONTROL  \
0                -1.372             -0.699      NaN      NaN    Male    False   
1                -1.091             -0.883      NaN      NaN    Male    False   
2                -1.017             -0.802      NaN      NaN    Male    False   
3                -0.144             -0.794      NaN      NaN     NaN    False   
4                -0.697             -0.472    Right     Left    Male    False   
..                  ...                ...      ...      ...     ...      ...   
498              -0.920             -0.633      NaN      NaN  Female    False   
499              -1.782             -1.761      NaN      NaN  Female    False   
500              -1.044             -1.006    Right     Left    Male    False   
501              -1.366             -1.230    Right     Left  Female    False   
502              -1.313             -1.184    Right     Left  Female    False   

                                                 NOTES  
0    Pegboard test, right hand--Some pegs were remo...  
1                                                  NaN  
2    Pegboard test, right hand--Some pegs were remo...  
3                                                  NaN  
4                                                  NaN  
..                                                 ...  
498                                                NaN  
499                                                NaN  
500  Keyboard test, right hand--Times of key data a...  
501                                                NaN  
50

In [32]:
TAPPD['PATNO'].value_counts() # 표본 자체가 너무 작음

4102    75
3234    22
3076    21
3230    20
3220    20
3227    19
3218    18
3083    18
3433    17
3440    17
3223    17
3436    16
3078    16
3089    16
3228    16
4051    16
3435    15
3443    15
3439    14
3224    14
4059    14
3229    13
3077    13
4103    13
3233    12
3432     8
4096     7
3088     7
3448     5
3086     4
3225     3
3226     2
Name: PATNO, dtype: int64

In [33]:
UEXTREMITY # Baseline 자체가 존재하지 않음

REC_ID F_STATUS  PATNO EVENT_ID  PAG_NAME   INFODT  NQMOB37  NQMOB30  \
0    723072701        S   4101      V13  NQOLLEFS  01/2019        4        4   
1    723251801        S   3522      V14  NQOLLEFS  01/2019        5        5   
2    723412901        S   3203      V15  NQOLLEFS  01/2019        5        4   
3    723823401        S   3479      V13  NQOLLEFS  01/2019        5        5   
4    765526501        S   3173      V15  NQOLLEFS  01/2020        5        5   
..         ...      ...    ...      ...       ...      ...      ...      ...   
427  770246201        S   4033      V14  NQOLLEFS  02/2020        5        5   
428  770293601        S   4035      V14  NQOLLEFS  03/2020        5        5   
429  770345401        S   3013      V15  NQOLLEFS  12/2019        5        5   
430  770915501        S   3206      V16  NQOLLEFS  03/2020        5        5   
431  771895001        S  60045      V13  NQOLLEFS  03/2020        5        3   

     NQMOB26  NQMOB32  NQMOB25  NQMOB33  NQMOB31  NQMOB28 ORIG_ENTRY  \
0          4        4        2        4        4        4    01/2019   
1          5        5        5        5        5        5    01/2019   
2          5        5        5        5        5        5    01/2019   
3          5        5        5        5        5        5    01/2019   
4          4        5        4        5        3        5    01/2020   
..       ...      ...      ...      ...      ...      ...        ...   
427        5        5        5        5        5        5    04/2020   
428        5        5        5        5        5        5    04/2020   
429        5        5        5        5        5        5    04/2020   
430        5        5        5        5        5        5    04/2020   
431        4        4        5        5        3        5    06/2020   

               LAST_UPDATE  QUERY SITE_APRV  
0    2020-06-29 14:26:27.0    NaN   01/2019  
1    2020-06-25 15:26:13.0    NaN   01/2019  
2    2020-06-26 19:14:44.0    NaN   01/2019  
3    2020-06-24 19:05:57.0    NaN   01/2019  
4    2020-06-30 09:46:22.0    NaN   01/2020  
..                     ...    ...       ...  
427  2020-06-29 13:59:17.0    NaN   04/2020  
428  2020-06-29 14:02:47.0    NaN   04/2020  
429  2020-06-30 09:26:13.0    NaN   04/2020  
430  2020-06-26 19:15:09.0    NaN   04/2020  
431  2020-06-30 16:55:43.0    NaN   06/2020  

[432 rows x 18 columns]

In [34]:
EPWORTH = EPWORTH[EPWORTH['EVENT_ID']=='BL']

In [35]:
EPWORTH = EPWORTH.drop(['REC_ID', 'F_STATUS', 'PAG_NAME', 'INFODT','PTCGBOTH',
       'ORIG_ENTRY', 'LAST_UPDATE', 'QUERY', 'SITE_APRV'], axis=1)

In [36]:
EPWORTH = EPWORTH.dropna(how='any')
EPWORTH

PATNO EVENT_ID  ESS1  ESS2  ESS3  ESS4  ESS5  ESS6  ESS7  ESS8
0      3404       BL     1   1.0     1     0   3.0     0     1   0.0
1      3400       BL     2   2.0     1     3   3.0     0     1   0.0
2      3401       BL     1   1.0     0     1   2.0     0     0   0.0
3      3405       BL     0   0.0     0     2   0.0     0     0   0.0
4      3403       BL     3   3.0     0     0   2.0     0     0   0.0
...     ...      ...   ...   ...   ...   ...   ...   ...   ...   ...
8011   5014       BL     1   1.0     0     1   1.0     0     0   0.0
8013   5015       BL     2   2.0     2     1   2.0     0     0   0.0
8025   5016       BL     3   3.0     2     3   1.0     3     3   0.0
8087   5017       BL     0   1.0     0     0   1.0     0     0   0.0
8213  40767       BL     0   1.0     0     0   2.0     0     2   0.0

[1471 rows x 10 columns]

In [37]:
REM = REM[REM['EVENT_ID']=='BL']

In [38]:
REM = REM.drop(['REC_ID', 'F_STATUS', 'PAG_NAME', 'INFODT', 'PTCGBOTH', 'CNSOTH', 'CNSOTHCM',
                      'EVENT_ID', 'ORIG_ENTRY', 'LAST_UPDATE', 'QUERY', 'SITE_APRV'], axis=1)
REM = REM.dropna(how='any')
REM

PATNO  DRMVIVID  DRMAGRAC  DRMNOCTB  SLPLMBMV  SLPINJUR  DRMVERBL  \
0      3401         1         0       0.0       0.0       0.0       0.0   
1      3400         1         1       0.0       1.0       0.0       1.0   
2      3404         1         0       0.0       1.0       0.0       1.0   
3      3403         1         1       0.0       1.0       0.0       0.0   
4      3405         1         1       0.0       0.0       0.0       0.0   
...     ...       ...       ...       ...       ...       ...       ...   
8429   5012         1         1       1.0       0.0       0.0       1.0   
8619   5014         1         1       1.0       1.0       1.0       1.0   
8621   5015         1         1       1.0       0.0       0.0       1.0   
8634   5016         0         0       0.0       0.0       0.0       0.0   
8696   5017         0         0       0.0       1.0       0.0       0.0   

      DRMFIGHT  DRMUMV  DRMOBJFL  ...  DRMREMEM  SLPDSTRB  STROKE  HETRA  \
0          0.0     0.0       0.0  ...       1.0       0.0       0    0.0   
1          1.0     1.0       1.0  ...       1.0       0.0       0    0.0   
2          0.0     0.0       0.0  ...       0.0       0.0       0    0.0   
3          0.0     0.0       0.0  ...       0.0       1.0       0    0.0   
4          0.0     0.0       0.0  ...       0.0       0.0       0    0.0   
...        ...     ...       ...  ...       ...       ...     ...    ...   
8429       1.0     0.0       0.0  ...       0.0       0.0       0    0.0   
8619       1.0     1.0       1.0  ...       0.0       1.0       0    0.0   
8621       0.0     0.0       0.0  ...       1.0       1.0       0    0.0   
8634       0.0     0.0       0.0  ...       0.0       0.0       0    0.0   
8696       0.0     0.0       0.0  ...       0.0       0.0       0    0.0   

      PARKISM  RLS  NARCLPSY  DEPRS  EPILEPSY  BRNINFM  
0           0  0.0       0.0    0.0       0.0      0.0  
1           1  0.0       0.0    1.0       0.0      0.0  
2           0  0.0       0.0    0.0       0.0      0.0  
3           1  0.0       0.0    0.0       0.0      0.0  
4           0  0.0       0.0    0.0       0.0      0.0  
...       ...  ...       ...    ...       ...      ...  
8429        1  0.0       0.0    0.0       0.0      0.0  
8619        1  0.0       0.0    0.0       0.0      0.0  
8621        1  0.0       0.0    0.0       0.0      0.0  
8634        1  0.0       0.0    0.0       0.0      0.0  
8696        1  0.0       0.0    0.0       0.0      0.0  

[1919 rows x 21 columns]

## 요약

* ARM: 데이터 크기가 단 26개 뿐. 활용 불가.
* LEXTREMITY: BL 자체가 존재하지 않는 데이터
* MDS1, MDS1Q, MDS2Q: 모두 활용 가능. 1923-4개 데이터
* SCHWAB: 활용 가능. 1727개 데이터
* HOUSEHOLD: 애매. 457개 데이터.
* LEISURE: 데이터 자체가 통합이 곤란. 차후 가능성 타진.
* TAPPD: 일부 극소수 환자만을 대상으로 중복. 활용 불가.
* UEXTREMITY: BL 자체가 존재하지 않는 데이터
* EPWORTH: 활용가능. 1471개 데이터
* REM: 활용가능. 1919개 데이터

In [39]:
MDS1

PATNO  NUPSOURC  NP1COG  NP1HALL  NP1DPRS  NP1ANXS  NP1APAT  NP1DDS
1       3400         1       0        0        2      3.0      2.0     0.0
3       3404         1       0        0        0      0.0      0.0     0.0
6       3401         1       0        0        0      0.0      0.0     0.0
7       3403         1       0        0        0      0.0      0.0     0.0
8       3405         1       0        0        0      0.0      0.0     0.0
...      ...       ...     ...      ...      ...      ...      ...     ...
13329   5012         1       1        0        0      1.0      2.0     0.0
13592   5014         1       0        0        0      1.0      0.0     0.0
13594   5015         1       2        0        0      0.0      1.0     0.0
13613   5016         1       2        0        1      1.0      0.0     0.0
13701   5017         1       0        0        0      0.0      0.0     0.0

[1923 rows x 8 columns]

In [40]:
MDS1Q

PATNO  NUPSOURC  NP1SLPN  NP1SLPD  NP1PAIN  NP1URIN  NP1CNST  NP1LTHD  \
2       3400         1      0.0      1.0      2.0      1.0      1.0      1.0   
3       3404         1      1.0      1.0      0.0      1.0      0.0      0.0   
4       3401         1      0.0      0.0      0.0      0.0      0.0      0.0   
7       3405         1      0.0      0.0      0.0      0.0      0.0      0.0   
8       3403         1      2.0      2.0      3.0      1.0      2.0      0.0   
...      ...       ...      ...      ...      ...      ...      ...      ...   
13331   5012         1      0.0      1.0      0.0      3.0      3.0      1.0   
13595   5014         1      2.0      0.0      2.0      3.0      2.0      1.0   
13598   5015         1      1.0      2.0      3.0      0.0      0.0      0.0   
13616   5016         1      0.0      0.0      1.0      0.0      0.0      0.0   
13703   5017         1      1.0      0.0      0.0      0.0      0.0      0.0   

       NP1FATG  
2          4.0  
3          1.0  
4          0.0  
7          0.0  
8          1.0  
...        ...  
13331      1.0  
13595      1.0  
13598      1.0  
13616      0.0  
13703      0.0  

[1924 rows x 9 columns]

In [41]:
MDS2Q

PATNO  NUPSOURC  NP2SPCH  NP2SALV  NP2SWAL  NP2EAT  NP2DRES  NP2HYGN  \
0       3401         1        0      0.0      0.0     0.0      0.0      0.0   
5       3404         1        0      0.0      0.0     0.0      0.0      0.0   
6       3400         1        0      2.0      0.0     2.0      1.0      1.0   
7       3405         1        0      0.0      0.0     0.0      0.0      0.0   
8       3403         1        0      0.0      0.0     1.0      1.0      0.0   
...      ...       ...      ...      ...      ...     ...      ...      ...   
13333   5012         1        0      3.0      0.0     1.0      2.0      0.0   
13596   5014         1        1      0.0      0.0     0.0      1.0      0.0   
13599   5015         1        1      3.0      1.0     0.0      1.0      1.0   
13617   5016         1        0      0.0      0.0     0.0      0.0      0.0   
13705   5017         1        0      0.0      0.0     0.0      0.0      0.0   

       NP2HWRT  NP2HOBB  NP2TURN  NP2TRMR  NP2RISE  NP2WALK  NP2FREZ  
0          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
5          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
6          1.0      2.0      1.0      1.0      2.0      1.0      1.0  
7          0.0      0.0      0.0      0.0      0.0      0.0      0.0  
8          1.0      1.0      1.0      0.0      2.0      1.0      0.0  
...        ...      ...      ...      ...      ...      ...      ...  
13333      3.0      4.0      2.0      2.0      3.0      1.0      2.0  
13596      0.0      0.0      1.0      0.0      1.0      1.0      0.0  
13599      1.0      1.0      1.0      1.0      2.0      3.0      0.0  
13617      0.0      4.0      0.0      2.0      4.0      3.0      0.0  
13705      0.0      0.0      0.0      1.0      0.0      0.0      0.0  

[1924 rows x 15 columns]

In [42]:
SCHWAB

PATNO  MSEADLG
3       3400       90
5       3403       80
6       3406       80
8       3407       90
11      3409       90
...      ...      ...
12172   5012       50
12414   5014      100
12416   5015       80
12437   5016       80
12519   5017      100

[1727 rows x 2 columns]

In [43]:
HOUSEHOLD

PATNO  LTHSWRK  HVYHSWRK  HMREPR  LAWNWRK  OUTGARDN  CAREGVR  WRKVL  \
44     3134        2         2       1        2       1.0      2.0    2.0   
76    60015        2         2       2        2       2.0      1.0    2.0   
168   12224        2         2       1        2       2.0      1.0    2.0   
217   14281        2         2       1        1       1.0      1.0    2.0   
241   60023        2         2       1        1       1.0      2.0    2.0   
...     ...      ...       ...     ...      ...       ...      ...    ...   
4977  41513        1         1       1        1       1.0      1.0    2.0   
5025  40750        1         1       1        1       1.0      1.0    2.0   
5035  40767        2         2       1        2       2.0      1.0    2.0   
5042  43086        2         1       1        1       1.0      2.0    1.0   
5043  42381        2         1       1        1       2.0      1.0    2.0   

      WRKVLHR  WRKVLACT  
44       50.0       3.0  
76        2.0       3.0  
168       2.0       2.0  
217       2.0       1.0  
241      12.0       1.0  
...       ...       ...  
4977     30.0       1.0  
5025     20.0       1.0  
5035     50.0       4.0  
5042     40.0       1.0  
5043      3.0       1.0  

[457 rows x 10 columns]

In [44]:
EPWORTH

PATNO EVENT_ID  ESS1  ESS2  ESS3  ESS4  ESS5  ESS6  ESS7  ESS8
0      3404       BL     1   1.0     1     0   3.0     0     1   0.0
1      3400       BL     2   2.0     1     3   3.0     0     1   0.0
2      3401       BL     1   1.0     0     1   2.0     0     0   0.0
3      3405       BL     0   0.0     0     2   0.0     0     0   0.0
4      3403       BL     3   3.0     0     0   2.0     0     0   0.0
...     ...      ...   ...   ...   ...   ...   ...   ...   ...   ...
8011   5014       BL     1   1.0     0     1   1.0     0     0   0.0
8013   5015       BL     2   2.0     2     1   2.0     0     0   0.0
8025   5016       BL     3   3.0     2     3   1.0     3     3   0.0
8087   5017       BL     0   1.0     0     0   1.0     0     0   0.0
8213  40767       BL     0   1.0     0     0   2.0     0     2   0.0

[1471 rows x 10 columns]

In [45]:
REM

PATNO  DRMVIVID  DRMAGRAC  DRMNOCTB  SLPLMBMV  SLPINJUR  DRMVERBL  \
0      3401         1         0       0.0       0.0       0.0       0.0   
1      3400         1         1       0.0       1.0       0.0       1.0   
2      3404         1         0       0.0       1.0       0.0       1.0   
3      3403         1         1       0.0       1.0       0.0       0.0   
4      3405         1         1       0.0       0.0       0.0       0.0   
...     ...       ...       ...       ...       ...       ...       ...   
8429   5012         1         1       1.0       0.0       0.0       1.0   
8619   5014         1         1       1.0       1.0       1.0       1.0   
8621   5015         1         1       1.0       0.0       0.0       1.0   
8634   5016         0         0       0.0       0.0       0.0       0.0   
8696   5017         0         0       0.0       1.0       0.0       0.0   

      DRMFIGHT  DRMUMV  DRMOBJFL  ...  DRMREMEM  SLPDSTRB  STROKE  HETRA  \
0          0.0     0.0       0.0  ...       1.0       0.0       0    0.0   
1          1.0     1.0       1.0  ...       1.0       0.0       0    0.0   
2          0.0     0.0       0.0  ...       0.0       0.0       0    0.0   
3          0.0     0.0       0.0  ...       0.0       1.0       0    0.0   
4          0.0     0.0       0.0  ...       0.0       0.0       0    0.0   
...        ...     ...       ...  ...       ...       ...     ...    ...   
8429       1.0     0.0       0.0  ...       0.0       0.0       0    0.0   
8619       1.0     1.0       1.0  ...       0.0       1.0       0    0.0   
8621       0.0     0.0       0.0  ...       1.0       1.0       0    0.0   
8634       0.0     0.0       0.0  ...       0.0       0.0       0    0.0   
8696       0.0     0.0       0.0  ...       0.0       0.0       0    0.0   

      PARKISM  RLS  NARCLPSY  DEPRS  EPILEPSY  BRNINFM  
0           0  0.0       0.0    0.0       0.0      0.0  
1           1  0.0       0.0    1.0       0.0      0.0  
2           0  0.0       0.0    0.0       0.0      0.0  
3           1  0.0       0.0    0.0       0.0      0.0  
4           0  0.0       0.0    0.0       0.0      0.0  
...       ...  ...       ...    ...       ...      ...  
8429        1  0.0       0.0    0.0       0.0      0.0  
8619        1  0.0       0.0    0.0       0.0      0.0  
8621        1  0.0       0.0    0.0       0.0      0.0  
8634        1  0.0       0.0    0.0       0.0      0.0  
8696        1  0.0       0.0    0.0       0.0      0.0  

[1919 rows x 21 columns]

In [46]:
STATUS

PATNO RECRUITMENT_CAT IMAGING_CAT ENROLL_DATE ENROLL_CAT ENROLL_STATUS  \
0     10362        PRODROMA    PRODROMA         NaN        NaN      Excluded   
1     10405        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2     10496        PRODROMA    PRODROMA         NaN        NaN      Excluded   
3     10606        PRODROMA    no image         NaN        NaN      Declined   
4     10662        PRODROMA    PRODROMA         NaN        NaN      Excluded   
...     ...             ...         ...         ...        ...           ...   
2246  92490        PRODROMA    PRODROMA     01/2015   PRODROMA      Enrolled   
2247  92516        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2248  92560        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2249  92834        PRODROMA    PRODROMA     02/2015   PRODROMA      Enrolled   
2250  92876        PRODROMA    PRODROMA         NaN        NaN      Excluded   

     DESCRP_CAT STATUS_DATE  
0           HYP     08/2013  
1           HYP     09/2014  
2           HYP     06/2013  
3           HYP     08/2013  
4           HYP     06/2014  
...         ...         ...  
2246        HYP     01/2015  
2247        HYP     11/2014  
2248        HYP     06/2014  
2249        HYP     02/2015  
2250        HYP     11/2014  

[2251 rows x 8 columns]

In [47]:
#pd.set_option('display.max_rows', None)
pd.reset_option('display.max_rows')
STATUS[STATUS['ENROLL_CAT'] != STATUS['RECRUITMENT_CAT']]

PATNO RECRUITMENT_CAT IMAGING_CAT ENROLL_DATE ENROLL_CAT ENROLL_STATUS  \
0     10362        PRODROMA    PRODROMA         NaN        NaN      Excluded   
1     10405        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2     10496        PRODROMA    PRODROMA         NaN        NaN      Excluded   
3     10606        PRODROMA    no image         NaN        NaN      Declined   
4     10662        PRODROMA    PRODROMA         NaN        NaN      Excluded   
...     ...             ...         ...         ...        ...           ...   
2243  91548        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2245  92138        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2247  92516        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2248  92560        PRODROMA    PRODROMA         NaN        NaN      Excluded   
2250  92876        PRODROMA    PRODROMA         NaN        NaN      Excluded   

     DESCRP_CAT STATUS_DATE  
0           HYP     08/2013  
1           HYP     09/2014  
2           HYP     06/2013  
3           HYP     08/2013  
4           HYP     06/2014  
...         ...         ...  
2243        HYP     03/2014  
2245        HYP     11/2014  
2247        HYP     11/2014  
2248        HYP     06/2014  
2250        HYP     11/2014  

[398 rows x 8 columns]

In [48]:
STATUS = STATUS.drop(['IMAGING_CAT', 'ENROLL_DATE', 'ENROLL_CAT', 'ENROLL_STATUS', 'DESCRP_CAT', 'STATUS_DATE'], axis=1)

In [49]:
STATUS

PATNO RECRUITMENT_CAT
0     10362        PRODROMA
1     10405        PRODROMA
2     10496        PRODROMA
3     10606        PRODROMA
4     10662        PRODROMA
...     ...             ...
2246  92490        PRODROMA
2247  92516        PRODROMA
2248  92560        PRODROMA
2249  92834        PRODROMA
2250  92876        PRODROMA

[2251 rows x 2 columns]

In [50]:
STATUS['RECRUITMENT_CAT'].value_counts()

PD          570
GENUN       445
GENPD       299
REGUN       253
HC          241
PRODROMA    216
REGPD       208
AV133        19
Name: RECRUITMENT_CAT, dtype: int64

In [51]:
data = pd.merge(STATUS, MDS1, how='inner', on='PATNO')

In [52]:
data = pd.merge(data, MDS1Q, how='inner', on='PATNO')

In [53]:
data = pd.merge(data, MDS2Q, how='inner', on='PATNO')

In [54]:
data = pd.merge(data, EPWORTH, how='inner', on='PATNO')

In [55]:
data = pd.merge(data, REM, how='inner', on='PATNO')

In [56]:
data.dropna(how='any')['RECRUITMENT_CAT'].value_counts()

PD          488
GENUN       423
GENPD       273
HC          196
PRODROMA     64
AV133        16
REGPD         1
Name: RECRUITMENT_CAT, dtype: int64

In [57]:
pd.merge(STATUS, HOUSEHOLD, how='inner', on='PATNO')['RECRUITMENT_CAT'].value_counts()

GENUN       293
GENPD       129
PRODROMA     33
PD            1
REGPD         1
Name: RECRUITMENT_CAT, dtype: int64

In [58]:
pd.merge(STATUS, SCHWAB, how='inner', on='PATNO')['RECRUITMENT_CAT'].value_counts()

PD          496
GENUN       427
GENPD       274
REGUN       249
REGPD       200
PRODROMA     64
AV133        16
HC            1
Name: RECRUITMENT_CAT, dtype: int64

SCHWAB과 HOUSEHOLD의 경우, 데이터 통합 시 HC에 해당하는 데이터가 전부 삭제되어버림

In [59]:
data = data.dropna(how='any')
data

PATNO RECRUITMENT_CAT  NUPSOURC_x  NP1COG  NP1HALL  NP1DPRS  NP1ANXS  \
0     10874        PRODROMA           1       0        0        0      0.0   
1     12224        PRODROMA           1       0        0        0      1.0   
2     12499        PRODROMA           1       0        0        0      0.0   
3     12593        PRODROMA           1       1        1        0      0.0   
4     13039        PRODROMA           1       0        0        0      0.0   
...     ...             ...         ...     ...      ...      ...      ...   
1456  90456        PRODROMA           1       1        0        0      1.0   
1457  91097        PRODROMA           1       0        0        0      0.0   
1458  91837        PRODROMA           1       0        0        0      0.0   
1459  92490        PRODROMA           1       1        0        0      0.0   
1460  92834        PRODROMA           1       0        0        1      0.0   

      NP1APAT  NP1DDS  NUPSOURC_y  ...  DRMREMEM  SLPDSTRB  STROKE  HETRA  \
0         0.0     0.0           1  ...       0.0       0.0       0    0.0   
1         0.0     0.0           1  ...       1.0       0.0       0    1.0   
2         0.0     0.0           1  ...       1.0       0.0       0    1.0   
3         0.0     0.0           1  ...       1.0       0.0       0    0.0   
4         0.0     0.0           1  ...       0.0       0.0       0    0.0   
...       ...     ...         ...  ...       ...       ...     ...    ...   
1456      0.0     0.0           1  ...       0.0       0.0       0    0.0   
1457      0.0     0.0           1  ...       0.0       0.0       1    0.0   
1458      0.0     0.0           1  ...       1.0       0.0       0    0.0   
1459      2.0     0.0           1  ...       1.0       0.0       0    0.0   
1460      1.0     0.0           1  ...       0.0       1.0       0    0.0   

      PARKISM  RLS  NARCLPSY  DEPRS  EPILEPSY  BRNINFM  
0           0  0.0       0.0    0.0       0.0      0.0  
1           0  0.0       0.0    1.0       0.0      0.0  
2           0  0.0       0.0    0.0       0.0      0.0  
3           0  0.0       0.0    0.0       0.0      0.0  
4           0  0.0       0.0    0.0       0.0      0.0  
...       ...  ...       ...    ...       ...      ...  
1456        0  0.0       0.0    0.0       0.0      0.0  
1457        0  0.0       0.0    1.0       0.0      0.0  
1458        0  0.0       0.0    0.0       0.0      0.0  
1459        0  0.0       0.0    0.0       0.0      0.0  
1460        0  0.0       0.0    0.0       0.0      0.0  

[1461 rows x 60 columns]

In [61]:
data.to_csv("Parkinson_Data_Combined.csv")